python/mpt/src/mpt/hash_diff.cairo

```diff
@@ -47,3 +59,97 @@ func poseidon_storage_diff{poseidon_ptr: PoseidonBuiltin*}(diff: StorageDiffEntr
     let (storage_diff_hash) = poseidon_hash_many(5, buffer);
     return storage_diff_hash;
 }
+
+func hash_account_diff_segment{poseidon_ptr: PoseidonBuiltin*}(account_diff: AccountDiff) -> felt {
+    alloc_locals;
+    let len = account_diff.value.len;
+    if (len == 0) {
+        // TODO: Do we return 0 as the hash of an empty segment?
+        // We don't want to raise here as a block could have 0 account diffs.
+        return 0;
+    }
+    let accumulator = poseidon_account_diff(account_diff.value.data[0]);
```

Comment:
shouldn't accumulator start at 0 ?

Reply:
why would we hash 0?

Reply:
I propose to define it as a recurrent sequence: 
u_0 = poseidon_account_diff(account_diff.value.data[0]);
u_n+1=poseidon(u_n, poseidon_account_diff(account_diff.value.data[n+1]))

Reply:
we don't hash 0
initial_acc is 0
then you start the loop at i=0

Reply:
but then the inner function has a different behavior for i == 0?
since it can't use the recursive construct "poseidon_hash_many(acc, new)"